src/CNLoginServer.cpp

```diff
@@ -112,6 +112,12 @@ void CNLoginServer::login(CNSocket* sock, CNPacketData* data) {
         userPassword = std::string(U16toU8(login->szPassword).c_str());
     }
 
+    // the client inserts a "\n" in the password if you press enter key in the middle of the password
+    // (not at the start or the end of the password field)
+    int i = userPassword.find("\n");
+    if (i > 0)
+        userPassword.erase(i, 1);
```

Comment:
I'd use a less obscure variable name here:
```suggestion
    int newlineIndex = userPassword.find("\n");
    if (newlineIndex > 0)
        userPassword.erase(newlineIndex, 1);
```